qt-models/cylindermodel.cpp

```diff
@@ -577,7 +577,8 @@ void CylindersModel::remove(const QModelIndex &index)
 	changed = true;
 	endRemoveRows();
 	cylinder_renumber(&displayed_dive, mapping);
-	DivePlannerPointsModel::instance()->cylinderRenumber(mapping);
+	if (in_planner())
+		DivePlannerPointsModel::instance()->cylinderRenumber(mapping);
```

Comment:
this sounds logical. I wonder why we didn't do this before. And is there a negative impact of doing this?
i.e., are you fixing a bug that caused you a problem, or just something in the code that seems wrong?

Reply:
DivePlannerPointsModel::instance()->cylinderRenumber(mapping);
was just created by me a couple of days before here:
73d2ab8099ce08ab20f119a126334463ffd41b1e
Before I did this we didn't care about correcting the cylinderid of entered planner dive data points at all which lead to bad effects when deleting a cylinder.
Fixing now my own bug because I forgot that at least CylindersModel::remove can be also called when not in the planner.
But your right... I'm doing things sometimes too quickly at the moment.

Reply:
You are doing a great job identifying issues and fixing them. I wish we had more people doing that.
I should have looked where this came from. Yes, of course, fixing your own new code should be merged. Please just merge your own PR in this case.